fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/metadata/Index.java

```diff
@@ -524,4 +525,27 @@ public String toString() {
         return str.toString();
     }
 
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        } else if (o == null || !getClass().equals(o.getClass())) {
+            return false;
+        }
+        Index that = (Index) o;
+        return this.name.equals(that.name)
+                && this.type.equals(that.type)
+                && this.rootExpression.equals(that.rootExpression)
+                && this.subspaceKey.equals(that.subspaceKey)
+                && this.addedVersion == that.addedVersion
+                && this.lastModifiedVersion == that.lastModifiedVersion
+                && Arrays.equals(this.primaryKeyComponentPositions, that.primaryKeyComponentPositions)
```

Comment:
Do we really want the `equals()` to compare `primaryKeyComponentPositions`? I'm not entirely sure what that's for but it's not included in the `toString()`.

Reply:
The primary key component positions are used to avoid duplicating primary key fields if there is overlap between the indexed fields and the primary key. (So, if I have an index of `(a, b)` and have primary key `(a, c)`, we want to create entries that look like `(a, b, c)` rather than `(a, b, a, c)` to save space, so the primary key components positions would be `[0, -1]` to indicate that the first component of the primary key is column 0 of the index entry and the second component is not contained in the entry at all.)

I think including this in `toString` would be excessive. I could also see removing it from equality checking, but two equal indexes really should have the same primary key component positions (I think).